pkg/machines/components/deleteDialog.jsx

```diff
@@ -39,7 +39,19 @@ const DeleteDialogBody = ({ disks, destroy, onChange }) => {
                                 onChange(index, event.target.checked);
                             }} />
                         <strong>{disk.target}</strong>
-                        <div className='disk-file'>{disk.file}</div>
+                        {disk.type == 'file' &&
+                        <div className='disk-source'>
+                            <label htmlFor='disk-source-file'> {_("Path")} </label>
+                            <span id='disk-source-file'> {disk.source.file} </span>
```

Comment:
`<label for="">` is supposed to apply to form elements only (and only certain ones that are not self-labelling). [See **WCAG 2.0 H44**](https://www.w3.org/TR/WCAG20-TECHS/H44.html#H44-description), where _note 1_ has a list of types of elements that get label (input of type text, checkbox, radio, file, & password; textarea; and select). Note 2 has some example form elements that don't get labels (inputs of type submit or reset, image buttons, hidden inputs, and script buttons).

There's also [the labelable element chart in the HTML5 spec](https://www.w3.org/TR/html5/sec-forms.html#labelable-element), which details which elements get `<label>`.

So then, what's good for a list of definitions? A description list, right? (They used to be called definition lists, but changed `dl` to mean "definition list" and `dt` to mean "definition term" instead of "description title".) We could even style it inline, naturally.

I guess that then gets tricky: https://snook.ca/archives/html_and_css/definition-lists-v-tables — however, this is from 2012. It might've changed since then.

Then there's the discussion @ https://webaim.org/discussion/mail_thread?thread=7089

Still, WAI seems to recommend them @ https://www.w3.org/WAI/tutorials/page-structure/content/#description-lists with no notes about screen readers.

Anyway, I have a few ideas, but my stomach is demanding I go to lunch. I'll research this a bit further and come up with a solution when I get back.

Reply:
`<label for="">` can work with [`<output>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/output) apparently... (TIL `<output>', the opposite of `<input>`.)

I wonder if we should use `<output>` instead of `<span>` here... or would it be bending the rules a bit too?

Reply:
...Alternatively, we could use a table with `<th>` & `<td>`. :grin: But we should probably avoid that still. :wink:

Reply:
PatternFly styles `<output>` as `display: block;`, so I guess we'll have to override that in CSS somewhere, if we use it.

Reply:
I thought that the whole point was to get rid of the table elements because they don't behave nicely in narrow mode. :/

Reply:
OH! We're already in a label for the checkbox. We shouldn't have a nested label. So I take everything back. :man_facepalming:

Reply:
From https://html.spec.whatwg.org/multipage/forms.html#the-label-element:

> Content model:
>    Phrasing content, but with no descendant labelable elements unless it is the element's labeled control, and no descendant label elements.

In other words, `<label>` can have text and the form control it specifies, but it cannot have additional form controls nor can it have labels.

The phrasing content may include things like span, strong, lists, and things like that.

Sorry for so many comments, some contradictory... I'm learning about this too at the same time. (I don't think any single person knows all of the HTML spec in their heads. It's easy to learn, but hard to do 100% right without referring to the docs... and I think we're in corner-case territory.)

**New suggestion**: We simply use `<span>` and `<strong>` here, I suppose. At least we know screen readers would work with that, and the items would be styled up visually in a correct manner too, especially as both are inline.